src/main/scala/com/itechart/ny_accidents/service/metric/DayPeriodMetricService.scala

```diff
@@ -0,0 +1,52 @@
+package com.itechart.ny_accidents.service.metric
+
+import java.util.Calendar
+
+import com.itechart.ny_accidents.constants.GeneralConstants._
+import com.itechart.ny_accidents.entity.MergedData
+import com.itechart.ny_accidents.service.DayPeriodService
+import org.apache.spark.rdd.RDD
+
+object DayPeriodMetricService {
+  private val sunriseSchedule = DayPeriodService.sunrisesSunsets
+
+  def getFrequency(data: RDD[MergedData]): Map[String,Long] ={
+    val filteredData = data.collect()
+      .filter(_.accident.dateTime.isDefined)
+      .map(_.accident.dateTime.get).groupBy(date => (date.getDate, date.getMonth)).map(t => (countMorningAndEveningSeconds(t._2.head), checkAccidents(t._2.toSeq)))
```

Comment:
.head - what if list was empty?

Reply:
according to the logic of the metric list can't be empty,
list is created only if something is going to be put there